framework/src/main/java/org/checkerframework/common/returnsreceiver/FluentAPIGenerator.java

```diff
@@ -0,0 +1,104 @@
+package org.checkerframework.common.returnsreceiver;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/**
+ * Enum of supported fluent API generators. For such generators, the checker can automatically
+ * add @This annotations on method return types in the generated code.
+ */
+public enum FluentAPIGenerator {
+    /**
+     * The <a
+     * href="https://github.com/google/auto/blob/master/value/userguide/builders.md">AutoValue</a>
+     * framework.
+     */
+    AUTO_VALUE {
+
+        private final String AUTO_VALUE_BUILDER = getAutoValueBuilderCanonicalName();
+
+        @Override
+        public boolean returnsThis(AnnotatedTypeMirror.AnnotatedExecutableType t) {
+            ExecutableElement element = t.getElement();
+            Element enclosingElement = element.getEnclosingElement();
+            boolean inAutoValueBuilder =
+                    AnnotationUtils.getAnnotationByName(
```

Comment:
It is inconsistent that this code uses `getAnnotationByName` but the below code uses `containsSameByName`.